src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs

```diff
@@ -57,5 +58,7 @@ public class ViewLoadedParams
         //{
             
         //}
+
+        public IEnumerable<WorkspaceViewModel> WorkSpaces;
```

Comment:
Let's identify the specific requirements of this component and only expose an interface.  MSVS's extract interface may be helpful here.  

Further, let's expose a property that obtains the DynamoViewModel's list of WorkspaceViewModel's so that a plug-in can invoke this at other points in the application lifetime and yield correct results.